modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,566 @@
+  {
+    auto elem_angles = vertex_angles(*elem);
+    if (MooseUtils::absoluteFuzzyEqual(elem_angles.front().first, M_PI, 0.001))
+    {
+      bad_elems_rec.push_back(std::make_tuple(elem, elem_angles.front().second, false));
+      bad_elem_counter++;
+    }
+  }
+  unsigned int bad_elem_ext = 0;
+  std::set<subdomain_id_type> new_subdomain_ids;
+  for (unsigned i = 0; i < bad_elem_counter; i++)
+  {
+    std::vector<boundary_id_type> elem_bdry_container;
+    Elem * elem_0 = std::get<0>(bad_elems_rec[i]);
+    Elem * elem_1 = elem_0->neighbor_ptr(std::get<1>(bad_elems_rec[i]));
+    Elem * elem_2 =
+        elem_0->neighbor_ptr((std::get<1>(bad_elems_rec[i]) - 1) % elem_0->n_vertices());
+    if (elem_1 == nullptr && elem_2 == nullptr)
+    {
+      bad_elem_ext++;
+      mesh.get_boundary_info().boundary_ids(
+          elem_0, std::get<1>(bad_elems_rec[i]), elem_bdry_container);
+    }
+    else
+      paramError("input", "The input mesh has degenerate quad element before trimming.");
+
+    auto elem_block_id = elem_0->subdomain_id();
+    auto pt0 = elem_0->node_ptr((std::get<1>(bad_elems_rec[i]) + 1) % elem_0->n_vertices());
+    auto pt1 = elem_0->node_ptr((std::get<1>(bad_elems_rec[i]) + 2) % elem_0->n_vertices());
+    auto pt2 = elem_0->node_ptr((std::get<1>(bad_elems_rec[i]) + 3) % elem_0->n_vertices());
+    for (unsigned int j = 0; j < n_elem_extra_ids; j++)
+      exist_extra_ids[j] = elem_0->get_extra_integer(j);
+    mesh.delete_elem(elem_0);
+    Elem * elem_Tri3 = mesh.add_elem(new Tri3);
+    elem_Tri3->set_node(0) = pt0;
+    elem_Tri3->set_node(1) = pt1;
+    elem_Tri3->set_node(2) = pt2;
+    for (auto bdry_id : elem_bdry_container)
+      boundary_info.add_side(elem_Tri3, 2, bdry_id);
+    elem_Tri3->subdomain_id() = elem_block_id + tri_subdomain_id_shift;
+    new_subdomain_ids.emplace(elem_block_id + tri_subdomain_id_shift);
+    for (unsigned int j = 0; j < n_elem_extra_ids; j++)
+      elem_Tri3->set_extra_integer(j, exist_extra_ids[j]);
+  }
+  for (auto & nid : new_subdomain_ids)
+  {
+    const SubdomainName old_name = mesh.subdomain_name(nid - tri_subdomain_id_shift);
+    if (MooseMeshUtils::getSubdomainID(
+            (old_name.empty() ? (SubdomainName)(std::to_string(nid - tri_subdomain_id_shift))
+                              : old_name) +
+                "_" + _tri_elem_subdomain_name_suffix,
+            mesh) != Moose::INVALID_BLOCK_ID)
+      paramError("tri_elem_subdomain_name_suffix",
+                 "The new subdomain name already exists in the mesh.");
+    mesh.subdomain_name(nid) =
+        (old_name.empty() ? (SubdomainName)(std::to_string(nid - tri_subdomain_id_shift))
+                          : old_name) +
+        "_" + _tri_elem_subdomain_name_suffix;
+    mooseWarning("Degenerate QUAD elements have been converted into TRI elements with a new "
+                 "subdomain name: " +
+                 mesh.subdomain_name(nid) + ".");
+  }
+  mesh.prepare_for_use();
```

Comment:
this can be expensive.  let's move it up in generate() so we dont have to search for it